From 07069e2609b05f7cfb024e1d355dd96026416feb Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Fri, 21 Nov 2014 20:14:28 +0800
Subject: [PATCH 1/8] atomisp2: set 2 pixel per clock in online video

This patch set 2 pixel per clock in online video mode such as offline mode
to improve SP performance.

Change-Id: Ib1358e0dc88f388982260fc6e1012659c819863b
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-10154
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/29688
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Jukka O Kaartinen <jukka.o.kaartinen@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_cmd.c      |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 1f181f5..da000a2 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -4246,13 +4246,13 @@ static int css_input_resolution_changed(struct atomisp_sub_device *asd,
 	dev_dbg(asd->isp->dev, "css_input_resolution_changed to %ux%u\n",
 		ffmt->width, ffmt->height);
 
-	if (asd->continuous_mode->val) {
-		/* Note for all checks: ffmt includes pad_w+pad_h */
 #if defined(ISP2401_NEW_INPUT_SYSTEM)
-		atomisp_css_input_set_two_pixels_per_clock(asd, false);
+	atomisp_css_input_set_two_pixels_per_clock(asd, false);
 #else
-		atomisp_css_input_set_two_pixels_per_clock(asd, true);
+	atomisp_css_input_set_two_pixels_per_clock(asd, true);
 #endif
+	if (asd->continuous_mode->val) {
+		/* Note for all checks: ffmt includes pad_w+pad_h */
 		if (asd->run_mode->val == ATOMISP_RUN_MODE_VIDEO ||
 		    (ffmt->width >= 2048 || ffmt->height >= 1536)) {
 			/*
-- 
1.7.9.5

